Blish HUD/GameServices/Input/MouseManager.cs

```diff
@@ -0,0 +1,189 @@
+            }
+
+            // Handle mouse events blocked by the mouse hook
+            if (_mouseEvent != null) {
+                switch (_mouseEvent.EventType) {
+                    case MouseEventType.MouseMoved:
+                    case MouseEventType.RightMouseButtonPressed:
+                    case MouseEventType.RightMouseButtonReleased:
+                    case MouseEventType.MouseWheelScrolled:
+                        break;
+                    case MouseEventType.LeftMouseButtonPressed:
+                    case MouseEventType.LeftMouseButtonReleased:
+                        OnMouseEvent(_mouseEvent);
+                        GameService.Graphics.SpriteScreen.TriggerMouseInput(_mouseEvent.EventType, newMouseState);
+                        break;
+                    default:
+                        Logger.Debug("Got unsupported input {mouseDataMessage}.", _mouseEvent.EventType);
+                        break;
+                }
+
+                _mouseEvent = null;
+            }
+
+            // Handle mouse right pressed/released
+            if (State.RightButton != newMouseState.RightButton) {
+                if (newMouseState.RightButton == ButtonState.Pressed) {
+                    this.RightMouseButtonPressed?.Invoke(this, new MouseEventArgs(MouseEventType.RightMouseButtonPressed));
+                    GameService.Graphics.SpriteScreen.TriggerMouseInput(MouseEventType.RightMouseButtonPressed, newMouseState);
+                } else if (newMouseState.RightButton == ButtonState.Released) {
+                    this.RightMouseButtonReleased?.Invoke(this, new MouseEventArgs(MouseEventType.RightMouseButtonReleased));
+                    GameService.Graphics.SpriteScreen.TriggerMouseInput(MouseEventType.RightMouseButtonReleased, newMouseState);
+                }
+            }
+
+            // Handle mouse scroll
+            if (newMouseState.ScrollWheelValue != 0) {
+                this.MouseWheelScrolled?.Invoke(this, new MouseEventArgs(MouseEventType.MouseWheelScrolled));
+                GameService.Graphics.SpriteScreen.TriggerMouseInput(MouseEventType.MouseWheelScrolled, newMouseState);
+            }
+
+            State = newMouseState;
+        }
+
+        protected override bool HandleNewInput(IntPtr wParam, IntPtr lParam) {
+            var newEvent = new MouseEventArgs((MouseEventType)wParam, Marshal.PtrToStructure<MouseLLHookStruct>(lParam));
+
+            if (newEvent.EventType == MouseEventType.MouseMoved) return false;
+
+            if (_cameraDragging && newEvent.EventType == MouseEventType.RightMouseButtonReleased) {
+                _cameraDragging = false;
+            } else if (_hudFocused && !_hookOverride) {
+                _mouseEvent = newEvent;
+
+                #if !NOMOUSEHOOK
+                return newEvent.EventType != MouseEventType.LeftMouseButtonReleased;
+                #endif
+            } else if (newEvent.EventType == MouseEventType.RightMouseButtonPressed) {
```

Comment:
Confirm this is set under the correct circumstances.

Reply:
Actually, the `_cameraDragging` variable is being properly set.  The issue comes from the break statement on line 134 of MouseManager.cs which prevents the `RightMouseButtonPressed`, `RightMouseButtonReleased`, and `MouseWheelScrolled` events from being triggered when they've been intercepted by the mouse hook (i.e. when the mouse is over one of our UI elements).

Reply:
This was resolved in f8f1e5d